src/vos/vos_internal.h

```diff
@@ -570,6 +570,8 @@ struct vos_rec_bundle {
 	struct vos_krec_df	*rb_krec;
 	/** input record size */
 	daos_size_t		 rb_rsize;
+	/** global record size, needed for EC singv record */
+	daos_size_t		 rb_gsize;
```

Comment:
It seems to me that we'd store this in tree root instead of in each records (along with the rb_rsize), probably in another patch? Jeff, is this something in your plan?

Reply:
thx. The record size possibly with different value on different epoch right? not sure how to store it in tree root node?

Reply:
No, we support only constant single value size, and I don't think there is strong requirement to support variable size. Though, there could be some tricky implementation issues when akey is punched then re-created with different record size.

Reply:
For single value, the record size is per update so Xuezhao is correct that different epochs will have different record size.   You can replace a 1 byte record with a 1MB record, for example.  I don't see any reason not to support that.

For Array value, the record size is the size of the array index and it is stored in the root and cannot be changed even after a punch unless aggregation has removed it.

Reply:
Ok, I was thinking that we support only fixed single value record size, and I always intentionally keep same rsize for the same single value in tests... (do we have any tests explicitly updating sv with variable rsizes?)